security/api_key_authentication.rst

```diff
@@ -389,40 +316,42 @@ using the ``simple_preauth`` and ``provider`` keys respectively:
             <config>
                 <!-- ... -->
 
-                <firewall name="secured_area"
+                <provider name="api_key_user_provider" id="AppBundle\Security\ApiKeyUserProvider" />
+
+                <firewall name="main"
                     pattern="^/api"
                     stateless="true"
                     provider="api_key_user_provider"
                 >
-                    <simple-preauth authenticator="apikey_authenticator" />
+                    <simple-preauth authenticator="AppBundle\Security\ApiKeyAuthenticator" />
                 </firewall>
-
-                <provider name="api_key_user_provider" id="api_key_user_provider" />
             </config>
         </srv:container>
 
     .. code-block:: php
 
         // app/config/security.php
 
-        // ..
+        // ...
+        use AppBundle\Security\ApiKeyAuthenticator;
+        use AppBundle\Security\ApiKeyUserProvider;
 
         $container->loadFromExtension('security', array(
+            'providers' => array(
+                'api_key_user_provider'  => array(
```

Comment:
There are two white spaces before the arrow operator here -> `'api_key_user_provider'  => array(`